### DISCUSSION TITLE
Fail to connect with Netlify

### DISCUSSION THREAD

**Discussion by knupel** [planetscale/discussion#523] [Q&A]
Hello

I try to connect my React Next application between PlanetScale and Netlify, but I fail and the console return an 'internal server error 500'. In local configuration that's work fine.

I've following the tutorial https://planetscale.com/docs/tutorials/deploy-to-netlify

I create a new password, like is said for the case where the password is blurred.

In Netlify I copy for the value like : mysql://l2z8tqg7e0us84ikh14p:****************@aws.connect.psdb.cloud/animal?sslaccept=strict where the ***** is my new password generated.

But maybe I miss something ?

**Comment by janpio** [comment]
What console is returning an error 500?
Did you test your app successfully locally before dpeloying?

**Comment by knupel** [comment]
Yes, that's work fine locally. And the error returned is very basic, just "Internal server error : 500", but i'm not good to read the error. 
here the link to web site : https://nextreactlab.netlify.app/database and here the link to my github project https://github.com/knupel/next_react_lab/tree/main/src/app/database

**Reply by janpio** [reply]
So you app does a POST to `api/animal` when you click the button. That returns a 500, and `Internal Server Error` - which is never good. You should have logs inside Netlify that you can look at what was logged before it crashed, causing the 500.

I suggest you look those up and figure out what they say. I bet they will tell you something about https://github.com/knupel/next_react_lab/blob/main/pages/api/animal.ts being off in some way.

**Reply by knupel** [reply]
hmmmm... I try to find where find a log in Netlify but I found nothing on my site.  I found Edge function but nothing in it, the only I have it's `RCS data routing`but nothing happen in it, when I try to element in my database. 
So I don't know where found the netlify log.
When you write about my file `api/animal.ts` you think may be I must add log in it ?

**Reply by knupel** [reply]
<img width="1203" alt="Capture d’écran 2023-08-25 à 15 14 48" src="https://github.com/planetscale/discussion/assets/3332269/4528c85d-1c5e-4420-8a37-79a184270d3e">
Do I also have to do this integration in addition to what is indicated in the tutorial ? Because the tutorial, don't talk about this integration. Anyway when I connect by this way it's a same result `500 / Internal Server Error`:(

**Comment by knupel** [comment]
I find a part of the solution, but is not enought !!! Now that work on `Netlify` but not on `Vercel` 
My idea, it was installing my app on Vercel to see if that's work, like expected that don't work, but like your suggestion to look the log in Netlify, I find the log in Vercel. and see there is a problem with the database, but with information and a link to https://www.prisma.io/docs/guides/other/troubleshooting-orm/help-articles/vercel-caching-issue. So I add in my file `package.json` 

```
"scripts" {
    "postinstall": "prisma generate"
  }
```
And now that's work for `Netlify` with response `status: 200` but not on `Vercel` where the answer is `status: 405
Method Not Allowed` 

So it's not perfect but it's a progress. the fun part it's `Vercel` solution for a `Netlify` problem !!!!
